src/human_player_module.c

```diff
@@ -0,0 +1,391 @@
+    if (*pp_head == NULL)
+    {
+        printf("List is already empty.");
+        return ERR_EMPTY_LIST;
+    }
+    while (to_delete->next != NULL)
+    {
+        if (to_delete->card.color == card_to_be_removed->color && to_delete->card.name == card_to_be_removed->name)
+        {
+            prev->next = to_delete->next;
+            free(&(to_delete->card));
+            return true;
+        }
+        to_delete = to_delete->next;
+    }
+    return false;
+}
+
+/**
+ * @brief Function to map the human player input to Card type structure
+ * Should be called only after validation of the user input
+ * 
+ * If user input is 3R, It will be mapped as follows
+ *  card.color=Red, card.name=3
+ * 
+ * @param input Value entered by the human during his turn
+ * 
+ * @return Card_t* Contains the mapped card name and color values
+ * If Card_t is NULL it means the input is an invalid one
+ */
+Card_t *map_user_input(char *input)
+{
+
+    Card_t *temp_card = (Card_t *)malloc(sizeof(Card_t));
+    temp_card = NULL;
+
+    if (input[0] == 'R')
+    {
+        temp_card->color = RED;
+    }
+    else if (input[0] == 'G')
+    {
+        temp_card->color = GREEN;
+    }
+    else if (input[0] == 'B')
+    {
+        temp_card->color = BLUE;
+    }
+
+    else if (input[0] == 'Y')
+    {
+        temp_card->color = YELLOW;
+    }
+    else
+    {
+        return temp_card;
+    }
+    //To map the number part of the User input to the card name field
+    temp_card->name = input[0] - '0';
+    return temp_card;
+}
+/**
+ * @brief Function picks a particular card from the current human deck of cards
+ * Called when a particular card has to be selected for removing from a deck
+ * Frees the menory allocated to he temporary selected drop card input parameter
+ * Note: Function  must be called only if the card to be selected is VALID
+ * 
+ * @param pp_head :Pointer which points to the address of head of card list in the Deck
+ * @param card_to_be_matched :Card which has to be matched from the deck
+ * 
+ * @return Card_t* Card that has been matched with the given card value
+ *         NULL - Returns NULL if the given card does not match with any of the cards in the deck
+ */
+Card_t *pick_card_from_deck(Deck_t **pp_head, Card_t *card_to_be_matched)
+{
+    Deck_t *temp = *pp_head;
+    while (temp->next != NULL)
+    {
+        if (temp->card.color == card_to_be_matched->color && temp->card.name == card_to_be_matched->name)
+        {
+            return &temp->card;
+        }
+        temp = temp->next;
+    }
+    free(card_to_be_matched); //Check if it can be freed here
+    return NULL;
+}
+
+/**
+ * @brief Displays the card assigned to the human player on the console during Request card/
+ * Draw1 and Draw2 scenarios
+ * 
+ * @param assigned_card Card that has been added newly to the player's deck
+ */
+void show_cards_assigned(Card_t assigned_card)
+{
+    printf("The card from the draw pile is %d%d \n", assigned_card.color, assigned_card.name);
```

Comment:
Can update it as following, then will print as string. eg: (BLUE, 7).
printf("The card from the draw pile is %s%s \n", CARD_COLOR_STRING[assigned_card.color], CARD_NAME_STRING[assigned_card.name]);

Reply:
Noted. Thanks!